### ISSUE TITLE
Conflicting guidlines for handling multiple tracestate headers

### ISSUE THREAD

**Issue by mwear** [w3c/trace-context#325]
There is conflicting information about how to handle multiple tracestate headers in the spec.

In section [3.3.1.1 tracestate Header Field Values](https://w3c.github.io/trace-context/#tracestate-header-field-values) we say:

> Multiple tracestate headers are allowed. Values from multiple headers in incoming requests SHOULD be combined in a single header according to Field Order [RFC7230], and sent as a single header in an outgoing request.

In section [3.3.1.5 tracestate Limits](https://w3c.github.io/trace-context/#tracestate-limits) we say:

> There might be multiple tracestate headers in a single request according to RFC7230 section 3.2.2. Vendors may propagate them as they came, combine them into a single header, or split them into multiple headers differently, following the RFC specification.

So, in one case we say that multiple tracestate headers must be combined before sending, and in the other, we say they can be combined, or sent separately.

We should clarify what we expect to happen in the multiple header case, and make sure the spec is consistent. I also wonder if we need to mention this at all, and could just say, tracestate can be split into multiple headers and should be processed according to [RFC7230 Section 3.2.2](https://tools.ietf.org/html/rfc7230#section-3.2.2).

**Comment by yurishkuro** [comment]
not really a conflict (should != must), but I agree that the "SHOULD combine" is unnecessary because the receiver still MUST be  able to cope with repeated header.

**Comment by mwear** [comment]
Perhaps we can clean this up by combining the ideas from both sections, and mentioning it only once. I'd propose that we put the following under [3.3.1.1 tracestate Header Field Values](https://w3c.github.io/trace-context/#tracestate-header-field-values)

> Tracestate SHOULD be sent as a single header when possible, but MAY be split into multiple headers. When a request contains multiple tracestate headers they SHOULD be combined into a single header. Tracestate headers MUST be split or combined as specified by [RFC7329 Section 3.2.2 Field Order](https://tools.ietf.org/html/rfc7230#section-3.2.2)

If that sounds ok, I can get a PR together with those changes and we can handle the wordsmithing  there.

**Comment by yurishkuro** [comment]
sgtm, but I think in  the 2nd sentence we need MUST, not SHOULD. Actually, I would rephrase it "When receiving multiple `tracestate` headers, then MUST be combined into a single `tracestate` per RFC7329".

**Comment by mtwo** [comment]
With these changes, are we specifying that actors that pass network traffic but otherwise do not participate in a trace (load balancers, proxies, etc.) are compelled to merge two headers together?

**Comment by mwear** [comment]
No, for sending, we are specifying a SHOULD. For receiving (perhaps reading might be a better word), they MUST be combined into a single header per RFC7230.

**Comment by mtwo** [comment]
Makes sense!

**Comment by danielkhan** [comment]
Already addressed.
